src/pages/Tailwind/Swap/SwapPanel.tsx

```diff
@@ -0,0 +1,388 @@
+          <PrimaryButton title="Swap" state={PrimaryButtonState.Disabled} onClick={() => console.log('clicked')} />
+        </div>
+      </div>
+    )
+  }
+
+  const ConfirmingContent = () => {
+    return (
+      <div className="mt-4">
+        <PrimaryButton type={PrimaryButtonType.Gradient} title="Confirming" state={PrimaryButtonState.InProgress} />
+      </div>
+    )
+  }
+
+  const EnterAmountContent = () => {
+    return (
+      <div className="mt-4">
+        <PrimaryButton type={PrimaryButtonType.Gradient} title="Enter an amount" state={PrimaryButtonState.Disabled} />
+      </div>
+    )
+  }
+
+  const InsufficientBalanceContent = () => {
+    return (
+      <div className="mt-4">
+        <PrimaryButton
+          type={PrimaryButtonType.Gradient}
+          title="Insufficient Balance"
+          state={PrimaryButtonState.Disabled}
+        />
+      </div>
+    )
+  }
+
+  const RetryContent = () => {
+    return (
+      <div className="mt-4">
+        <RetryButton title="Retry" isEnabled={true} onClick={() => console.log('clicked')} />
+      </div>
+    )
+  }
+
+  const CurrentButtonContent = () => {
+    if (approveState === ApproveButtonState.NotApproved && buttonState === ButtonState.EnterAmount) {
+      if (!fromInputValue || !toInputValue || parseFloat(fromInputValue) === 0 || parseFloat(toInputValue) === 0) {
+        return <EnterAmountContent />
+      } else if (fromInputValue && parseFloat(fromInputValue) > 0 && toInputValue && parseFloat(toInputValue) > 0) {
+        return <NotApproveContent />
+      }
+    }
+    if (approveState === ApproveButtonState.Approving) {
+      return <ApprovingContent />
+    }
+    if (approveState === ApproveButtonState.Approved && buttonState === ButtonState.Default) {
+      return <ApprovedContent />
+    }
+    if (approveState === ApproveButtonState.Approved && buttonState === ButtonState.Swap) {
+      return <SwapContent />
+    }
+    if (approveState === ApproveButtonState.Approved && buttonState === ButtonState.Confirming) {
+      return <ConfirmingContent />
+    }
+    if (approveState === ApproveButtonState.Approved && buttonState === ButtonState.Retry) {
+      return <RetryContent />
+    }
+    if (approveState === ApproveButtonState.Approved && buttonState === ButtonState.InsufficientBalance) {
+      return <InsufficientBalanceContent />
+    }
+    return <></>
+  }
+
+  const MainContent = () => {
+    const toggleWalletModal = useWalletModalToggle()
+
+    if (!account && !error) {
+      return (
+        <div className="mt-2">
+          <ConnectButton title="Connect to Wallet" onClick={() => toggleWalletModal()} />
+        </div>
+      )
+    } else {
+      return (
+        <>
+          <CurrentButtonContent />
+          <SwapDetails
+            price={price}
+            toCurrency={toCurrency.symbol}
+            fromCurrency={fromCurrency.symbol}
+            minimumReceived={minimumAmount}
+          />
+        </>
+      )
+    }
```

Comment:
Could probably remove the else?
```suggestion
    }
    
    return (
        <>
          <CurrentButtonContent />
          <SwapDetails
            price={price}
            toCurrency={toCurrency.symbol}
            fromCurrency={fromCurrency.symbol}
            minimumReceived={minimumAmount}
          />
        </>
      )
    }
```